internal/pke/workflow/adapter/secret_store.go

```diff
@@ -0,0 +1,71 @@
+// Copyright © 2019 Banzai Cloud
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package adapter
+
+import (
+	"context"
+	"fmt"
+
+	"emperror.dev/errors"
+
+	"github.com/banzaicloud/pipeline/internal/pke/workflow"
+	pkgSecret "github.com/banzaicloud/pipeline/pkg/secret"
+	"github.com/banzaicloud/pipeline/secret"
+)
+
+type PasswordSecretStore struct {
+	store SecretStore
+}
+
+type SecretStore interface {
+	Get(orgID uint, secretID string) (*secret.SecretItemResponse, error)
```

Comment:
You could use the common secret store here. You only need the values, nothing else from the low level secret API.